platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java

```diff
@@ -154,4 +132,82 @@ Properties loadProperties(String location) {
         return PropertiesLoader.loadProperties(location);
     }
 
+    /**
+     * Runnable to refresh the CRL from the URL when either the nextUpdate time has elasped
+     * or it has rolled over to the next day.
+     */
+    private class CrlRefresh implements Runnable {
+
+        private Calendar start = Calendar.getInstance();
+
+        public void run() {
+            Properties encryptionProperties = loadProperties(encryptionPropertiesLocation);
+            CRL crl = crlCache.get();
+            if (crl instanceof X509CRL) {
+                X509CRL x509Crl = (X509CRL) crl;
+                if (Calendar.getInstance()
+                        .getTime()
+                        .after(x509Crl.getNextUpdate()) || rolledOverDay()) {
+                    setCrlLocation(encryptionProperties.getProperty(CRL_PROPERTY_KEY));
```

Comment:
It would be nice to have a log statement indicating that the CRL has been/is being updated. At least at the DEBUG level.